src/models.js

```diff
@@ -0,0 +1,188 @@
+class GraphFunc {
+    constructor (f='cos(2*x + 2*y)/(1 + x^2 + y^2)', rangeX=[-2, 2],
+                 rangeY=[-2, 2], meshX=10, meshY=10, res=30,
+                 lvlUp=true, viewGraph=true, rangeT=[0, 2], tangents=false,
+                 color='#000000', integrate=false, n=1, sample=0) {
+        this.f = f;
+        this.rangeX = rangeX;
+        this.rangeY = rangeY;
+        this.meshX = meshX;
+        this.meshY = meshY;
+        this.res = res;
+        this.lvlUp = lvlUp;
+        this.viewGraph = viewGraph;
+        this.rangeT = rangeT;
+        this.tangents = tangents;
+        this.color = color;
+        this.integrate = integrate;
+        this.n = n;
+        this.sample = sample;
+    }
+
+    type = 'GraphFunc';
+}
+
+class LevelSurface {
+    constructor (g='x^2 + y^2 - z^2', k=1, rangeX=[-2,2], rangeY=[-2, 2],
+                 rangeZ=[-2, 2], tangents=false) {
+        this.g = g;
+        this.k = k;
+        this.rangeX = rangeX;
+        this.rangeY = rangeY;
+        this.rangeZ = rangeZ;
+        this.tangents = tangents;
+    }
+
+    type = 'LevelSurface';
+}
+
+class Message {
+    constructor (message, sender) {
+        this.message = message;
+        this.sentBy = sender;
+    }
+
+    timestamp = Date.now();
+}
+
+class ObjectMessage extends Message {
+    constructor (message, objectList, sender) {
+        super(message, sender);
+        this.objects = Array.from(objectList);
+    }
+}
+
+class ParametricSurface {
+    constructor (x='cos(u)*(1 + sin(v)/3)', y='sin(u)*(1 + sin(v)/3)',
+                 z='cos(v)/3', rangeU=['0', '2*pi'], rangeV=['0', '2*pi'],
+                 meshU=10, meshV=10, res=30) {
+        this.x = x;
+        this.y = y;
+        this.z = z;
+        this.rangeU = rangeU;
+        this.rangeV = rangeV;
+        this.meshU = meshU;
+        this.meshV = meshV;
+        this.res = res;
+    }
+
+    type = 'ParametricSurface';
+}
+
+/**
+ * A basic container for holding and sorting responses.
+ */
+class ResponseList {
```

Comment:
I think a ResponseList abstraction may possibly be necessary, but not until we iron out the basics of what a Response is. I would replace this with a PollResponse class to represent a poll response.

Reply:
I was thinking that Message would be both a Question and Response. To that end I didn't include a message id because I was under the impression that there would only be one question open at a time. If we allow multiple open questions then there would be a message id to match student responses to a given question. A ResponseList would be generated to contain the Responses to that given question.

While we don't need it right now I thought it'd be good to keep in the back of our minds as we move forward in development.

I'll remove it if you feel that it's in the way.